pallets/xyk-liquidity-mining/src/lib.rs

```diff
@@ -0,0 +1,964 @@
+			global_farm_id: GlobalFarmId,
+			yield_farm_id: YieldFarmId,
+			who: T::AccountId,
+			asset_pair: AssetPair,
+		},
+
+		/// Yield farm multiplier was updated.
+		YieldFarmUpdated {
+			global_farm_id: GlobalFarmId,
+			yield_farm_id: YieldFarmId,
+			who: T::AccountId,
+			asset_pair: AssetPair,
+			multiplier: FarmMultiplier,
+		},
+
+		/// NFT representing deposit has been destroyed
+		DepositDestroyed { who: T::AccountId, deposit_id: DepositId },
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Create new liquidity mining program with provided parameters.
+		///
+		/// `owner` account has to have at least `total_rewards` balance. This fund will be
+		/// transferred from `owner` to farm account.
+		///
+		/// The dispatch origin for this call must be `T::CreateOrigin`.
+		/// !!!WARN: `T::CreateOrigin` has power over funds of `owner`'s account and it should be
+		/// configured to trusted origin e.g Sudo or Governance.
+		///
+		/// Parameters:
+		/// - `origin`: global farm's owner.
+		/// - `total_rewards`: total rewards planned to distribute. This rewards will be
+		/// distributed between all yield farms in the global farm.
+		/// - `planned_yielding_periods`: planned number of periods to distribute `total_rewards`.
+		/// WARN: THIS IS NOT HARD DEADLINE. Not all rewards have to be distributed in
+		/// `planned_yielding_periods`. Rewards are distributed based on the situation in the yield
+		/// farms and can be distributed in a longer time frame but never in the shorter time frame.
+		/// - `blocks_per_period`:  number of blocks in a single period. Min. number of blocks per
+		/// period is 1.
+		/// - `incentivized_asset`: asset to be incentivized in XYK pools. All yield farms added into
+		/// liq. mining program have to have `incentivized_asset` in their pair.
+		/// - `reward_currency`: payoff currency of rewards.
+		/// - `owner`: liq. mining program owner.
+		/// - `yield_per_period`: percentage return on `reward_currency` of all farms p.a.
+		/// - `min_deposit`: minimum amount which can be deposited to the farm
+		/// - `price_adjustment`:
+		/// Emits `GlobalFarmCreated` event when successful.
+		#[pallet::call_index(0)]
+		#[pallet::weight(<T as Config>::WeightInfo::create_global_farm())]
+		pub fn create_global_farm(
+			origin: OriginFor<T>,
+			total_rewards: Balance,
+			planned_yielding_periods: PeriodOf<T>,
+			blocks_per_period: BlockNumberFor<T>,
+			incentivized_asset: AssetId,
+			reward_currency: AssetId,
+			owner: T::AccountId,
+			yield_per_period: Perquintill,
+			min_deposit: Balance,
+			price_adjustment: FixedU128,
+		) -> DispatchResult {
+			<T as pallet::Config>::CreateOrigin::ensure_origin(origin)?;
+
+			if !T::AssetRegistry::is_sufficient(reward_currency) {
+				let ed =
+					T::AssetRegistry::existential_deposit(reward_currency).ok_or(Error::<T>::AssetNotRegistered)?;
+
+				let pot = T::LiquidityMiningHandler::pot_account().ok_or(Error::<T>::FailToGetPotId)?;
```

Comment:
As I understand we need to seed the pot with ED on farm creation. We have two pots though, the first one is seeded on transfer by the farm creator. We could put some comment about this fact.